### PR TITLE
Added a ceph dashboard proposal with cloud selector

### PR DESCRIPTION
**Pull request by josecastillolema**

### CODE REVIEW

**Comment by leifmadsen** [comment]
Thanks very much for the submission. We're getting to the point (finally :)) to have a base configuration to target dashboards against for STF 1.3. Hopefully we can test/refine/merge this in the next couple of weeks as we've been doing dashboard work recently.

**Comment by pleimer** [comment]
Can you wrap this in a GrafanaDashboard custom resource object like https://github.com/infrawatch/dashboards/blob/master/deploy/rhos-dashboard.yaml? It makes it easier to load and view changes

**Review by pleimer — CHANGES_REQUESTED** [review]
Thanks for this! Can you wrap this in a GrafanaDashboard custom resource object like https://github.com/infrawatch/dashboards/blob/master/deploy/rhos-dashboard.yaml? It makes it easier to load and view changes

**Comment by josecastillolema** [comment]
Done @pleimer !
Dashboard wrapped into a GrafanaDashboard custom resource object.

**Comment by leifmadsen** [comment]
Wow that was quick :)  Thanks! Maybe we can do a bit of testing this week and see how close it is.

**Comment by leifmadsen** [comment]
I had to do a bit of changing around because the format I have doesn't seem to align to what you have around the metrics. Overall the dashboard does look pretty good for the functionality that I have available to me and the data I'm passing.

Do you have a copy of your OSP configuration for the collectd Ceph plugin? You're using a few metrics/labels that I don't have access to, and I'd like to try and align our base configuration so that things all work out of the box. Thanks @josecastillolema !

**Comment by josecastillolema** [comment]
The OSP configuration file (from RHOSP13):
```
parameter_defaults:
  EventPipelinePublishers: []
  MetricPipelinePublishers: []
  PipelinePublishers: []
  ManagePipeline: true
  ManagePolling: true
  CeilometerEnablePanko: false
  CeilometerQdrPublishEvents: true
  CeilometerQdrPublishMetrics: true
  CeilometerQdrEventsConfig:
    driver: amqp
    topic: labnk-event
  CeilometerQdrMetricsConfig:
    driver: amqp
    topic: labnk-metering
  CollectdAmqpInstances:
    labnk-notify:
      format: JSON
      notify: true
      presettle: false
    labnk-telemetry:
      format: JSON
      presettle: false
  CollectdAmqpInterval: 30
  CollectdConnectionType: amqp1
  CollectdDefaultPlugins:
    - ceph
    - cpu
    - df
    - disk
    - hugepages
    - interface
    - load
    - memory
    - unixsock
    - ovs_stats
    - uptime
  CollectdDefaultPollingInterval: 30
  MetricsQdrAddresses:
    - distribution: multicast
      prefix: collectd
    - distribution: multicast
      prefix: anycast/ceilometer
  MetricsQdrConnectors:
    - host: default-interconnect-...
      port: 443
      role: edge
      sslProfile: sslProfile
      verifyHostname: false
  MetricsQdrSSLProfiles:
    - name: sslProfile
  ExtraConfig:
    ceilometer::agent::polling::polling_interval: 30
    collectd::plugin::cpu::reportbycpu: true
    collectd::plugin::cpu::reportbystate: true
    collectd::plugin::cpu::reportnumcpu: true
    collectd::plugin::cpu::valuespercentage: true
    collectd::plugin::df::ignoreselected: true
    collectd::plugin::df::reportbydevice: true
    collectd::plugin::df::fstypes: ['xfs']
    collectd::plugin::load::reportrelative: true
    collectd::plugin::virt::connection: "qemu:///system"
    collectd::plugin::virt::extra_stats: "cpu_util disk disk_err pcpu job_stats_background perf vcpupin vcpu memory domain_state"
    collectd::plugin::virt::hostname_format: "hostname name uuid"
    collectd::plugin::vmem::verbose: true
    collectd::plugin::virt::plugin_instance_format: metadata
    collectd::plugin::ovs_stats::socket: '/run/openvswitch/db.sock'
    collectd::plugin::ceph::daemons:
    - ceph-mon.lab01nkpcr003
    - ceph-mon.lab01nkpcr004
    - ceph-mon.lab01nkpcr006
    - ceph-osd.0
    - ceph-osd.1
    - ceph-osd.2
    - ceph-osd.3
    - ceph-osd.4
    - ceph-osd.5
    - ceph-osd.6
    - ceph-osd.7
    - ceph-osd.8
    - ceph-osd.9
    - ceph-osd.10
    - ceph-osd.11
    - ceph-osd.12
    - ceph-osd.13
    - ceph-osd.14
    - ceph-osd.15
```

**Comment by leifmadsen** [comment]
Confirmed this is working as-is once you have the MONs reporting data as well. I have an open issue to identify documentation that allows deployment of OSP in such a way that the `ceph` plugin is scheduled for deployment on the controller nodes. (See https://bugzilla.redhat.com/show_bug.cgi?id=1984193)

I might also go through an O'Reilly book I found about monitoring Ceph and see about enhancing the dashboard based on their recommendations, including adding CPU and memory load information for the Ceph services.

**Review by leifmadsen — APPROVED** [review]

File: contrib/ceph-dashboard.yaml [hunk 1]
```diff
@@ -0,0 +1,818 @@
...
+      "style": "dark",
+      "tags": [],
+      "templating": {
+        "list": [
+          {
+            "allValue": null,
+            "current": {
+              "selected": true,
+              "text": "default-labnk-coll-meter-smartgateway",
+              "value": "default-labnk-coll-meter-smartgateway"
```
Comment (leifmadsen) [inline contrib/ceph-dashboard.yaml:774]:
```suggestion
              "text": "",
              "value": ""
```

**Comment by leifmadsen** [comment]
I've made a few changes and updated according to some of our current approaches. I'm going to close this one in favour of #33 which is still based on your work.

**Pull request closed without merging**